packages/auth/e2e/multiFactor.e2e.js

```diff
@@ -0,0 +1,448 @@
+    });
+    it(':android: can enroll new factor without display name', async function () {
+      try {
+        await createVerifiedUser('verified@example.com', 'test123');
+        const phoneNumber = getRandomPhoneNumber();
+
+        should.deepEqual(firebase.auth().currentUser.multiFactor.enrolledFactors, []);
+        const multiFactorUser = await firebase.auth.multiFactor(firebase.auth());
+
+        const session = await multiFactorUser.getSession();
+
+        const verificationId = await firebase
+          .auth()
+          .verifyPhoneNumberForMultiFactor({ phoneNumber, session });
+        const verificationCode = await getLastSmsCode(phoneNumber);
+        const cred = firebase.auth.PhoneAuthProvider.credential(verificationId, verificationCode);
+        const multiFactorAssertion = firebase.auth.PhoneMultiFactorGenerator.assertion(cred);
+        await multiFactorUser.enroll(multiFactorAssertion);
+
+        const enrolledFactors = firebase.auth().currentUser.multiFactor.enrolledFactors;
+        enrolledFactors.length.should.equal(1);
+        should.equal(enrolledFactors[0].displayName, null);
+      } catch (e) {
+        return Promise.reject(e);
+      }
+      return Promise.resolve();
+    });
+    it(':android: can enroll multiple factors', async function () {
+      const { email, password, phoneNumber } = await createUserWithMultiFactor();
+      await signInUserWithMultiFactor(email, password, phoneNumber);
+
+      const anotherNumber = getRandomPhoneNumber();
+      const multiFactorUser = await firebase.auth.multiFactor(firebase.auth());
+
+      const session = await multiFactorUser.getSession();
+      const verificationId = await firebase
+        .auth()
+        .verifyPhoneNumberForMultiFactor({ phoneNumber: anotherNumber, session });
+      const verificationCode = await getLastSmsCode(anotherNumber);
+      const cred = firebase.auth.PhoneAuthProvider.credential(verificationId, verificationCode);
+      const multiFactorAssertion = firebase.auth.PhoneMultiFactorGenerator.assertion(cred);
+      const displayName = 'Another displayName';
+      await multiFactorUser.enroll(multiFactorAssertion, displayName);
+
+      const enrolledFactors = firebase.auth().currentUser.multiFactor.enrolledFactors;
+      enrolledFactors.length.should.equal(2);
+      const matchingFactor = enrolledFactors.find(factor => factor.displayName === displayName);
+      matchingFactor.should.be.an.Object();
+      matchingFactor.uid.should.be.a.String();
+      matchingFactor.enrollmentTime.should.be.a.String();
+      matchingFactor.factorId.should.equal('phone');
+
+      return Promise.resolve();
+    });
+    it(':android: can not enroll the same factor twice', async function () {
+      // This test should probably be implemented but doesn't work:
+      // Every time the same phone number requests a verification code,
+      // the emulator endpoint does not return a code, even though the emulator log
+      // prints a code.
```

Comment:
firebase-tools repo (which hosts the emulator) appears to be tracking this - I just chimed in over there: https://github.com/firebase/firebase-tools/issues/4290#issuecomment-1281260335 - could probably link it here

```suggestion
      // This test should probably be implemented but doesn't work:
      // Every time the same phone number requests a verification code,
      // the emulator endpoint does not return a code, even though the emulator log
      // prints a code.
      // See https://github.com/firebase/firebase-tools/issues/4290#issuecomment-1281260335
```